tests/unit/TechRecordsService.unitTest.ts

```diff
@@ -588,13 +656,54 @@ describe("updateTechRecord", () => {
         const techRecordsService = new TechRecordsService(mockDAO, s3Mock);
         const recordToUpdate: any = {vin: "123456656"};
         try {
-          expect(await techRecordsService.updateTechRecord(recordToUpdate, msUserDetails, ["nsa7zXuM/5iYmrCM2kzmT"])).toThrowError();
+          expect(await techRecordsService.updateTechRecord(recordToUpdate, msUserDetails, files)).toThrowError();
         } catch (errorResponse) {
           expect(errorResponse).toBeInstanceOf(HTTPError);
           expect(errorResponse.statusCode).toEqual(500);
-          expect(errorResponse.body).toEqual(HTTPRESPONSE.S3_ERROR);
+          expect(errorResponse.body).toEqual("The specified bucket does not exist.");
         }
       });
+
+      it("should skip the upload if the user doesn't upload any new files and return the existing uploaded files", async () => {
+        const techRecord: any = cloneDeep(records[29]);
+        techRecord.techRecord[0].adrDetails.documents = ["existingFile.pdf"];
+        const MockDAO = jest.fn().mockImplementation(() => {
+          return {
+            updateSingle: () => {
+              return Promise.resolve({
+                Attributes: techRecord
+              });
+            },
+            getBySearchTerm: () => {
+              return Promise.resolve({
+                Items: [cloneDeep(records[29])],
+                Count: 1,
+                ScannedCount: 1
+              });
+            }
+          };
+        });
+        const S3Mock = jest.fn().mockImplementation(() => {
+          return null;
+        });
+        const mockDAO = new MockDAO();
+        const s3Mock = new S3Mock();
+        const techRecordsService = new TechRecordsService(mockDAO, s3Mock);
+        const recordToUpdate: any = {
+          vin: techRecord.vin,
+          partialVin: techRecord.partialVin,
+          primaryVrm: techRecord.primaryVrm,
+          techRecord:
+            [{
+              reasonForCreation: techRecord.techRecord[0].reasonForCreation,
+              adrDetails: techRecord.techRecord[0].adrDetails
+            }]
+        };
+        const existingFiles = [{toUpload: false, filename: "existingFile.pdf", base64String: ""}];
+        const response: any = await techRecordsService.updateTechRecord(recordToUpdate, msUserDetails, existingFiles);
+        expect(response).toBeDefined();
```

Comment:
Not sure this test actually tests what it says it does. You can mock "uploadFile" or the S3BucketService.upload function and check that there are no invocations even if you pass a new file with toUpload = false

Reply:
fixed

Reply:
Perfect, cheers